### PR TITLE
refactor: support standalone compactor node

### PR DESCRIPTION
**Pull request by zwang28**
## What's changed and what's your intention?

This PR makes compactor as a dedicated process, like compute node/frontend/meta node.
- Add a `compactor` crate under `src/storage`.
- Modify risedev conf. As compactor node is incompatible with in-mem object store, we start a compactor tokio task for it.

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/2085

### CODE REVIEW

File: src/risedevtool/src/task/compactor_service.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            ))
+            .arg("--metrics-level")
+            .arg("1");
+
+        let provide_minio = config.provide_minio.as_ref().unwrap();
+        let provide_aws_s3 = config.provide_aws_s3.as_ref().unwrap();
+        match (provide_minio.as_slice(), provide_aws_s3.as_slice()) {
+            ([], []) => {
+                return Err(anyhow!(
+                    "Compactor is not compatible with in_memory_kv_state_backend. Need to enable either minio or aws-s3.",
```
Comment (skyzh) [inline src/risedevtool/src/task/compactor_service.rs:61]: `in_memory_kv_state_backend` -> in-memory state backend. There are two types of in-memory state backend: the BTreeMap KV based one, and the hummock + in-memory one.

**Review by skyzh — APPROVED** [review]
LGTM! I think we should also start compactor thread for hummock in-memory mode? Otherwise e2e will be significantly slower I guess.

File: src/compute/src/server.rs [hunk 2]
```diff
@@ -114,16 +113,6 @@ pub async fn compute_node_serve(
     .await
     .unwrap();
 
-    // A hummock compactor is deployed along with compute node for now.
-    if let Some(hummock) = state_store.as_hummock_state_store() {
-        sub_tasks.push(Compactor::start_compactor(
-            hummock.inner().options().clone(),
```
Comment (BugenZhao) [inline src/compute/src/server.rs:120]: We may also remove the inner method. 😄

Comment (zwang28) [reply src/compute/src/server.rs:120]: Still used in ssbench. The call site will be refactored and then we can remove inner.

**Comment by skyzh** [comment]
... also one suggestion: if compute node found that minio / s3 is enabled, but there are no compactors in cluster, then we should also report an error when starting risedev?

**Review by skyzh — APPROVED** [review]

**Pull request merged**